content/blog/aws-eks-managed-nodes-fargate/index.md

```diff
@@ -0,0 +1,356 @@
+            [0]: {
+                context: {
+                    cluster: "kubernetes"
+                    user   : "aws"
+                }
+                name   : "aws"
+            }
+        ]
+        current-context: "aws"
+        kind           : "Config"
+        users          : [
+            [0]: {
+                name: "aws"
+                user: {
+                    exec: {
+                        apiVersion: "client.authentication.k8s.io/v1alpha1"
+                        args      : [
+                            [0]: "token"
+                            [1]: "-i"
+                            [2]: "cluster-eksCluster-560236b"
+                        ]
+                        command   : "aws-iam-authenticator"
+                    }
+                }
+            }
+        ]
+    }
+
+Resources:
+    + 37 created
+
+Duration: 12m12s
+```
+
+## Manually Managing EC2 Node Groups
+
+A fully functioning production cluster typically requires many other considerations. For those, we've put together [a set of playbooks as part of Pulumi Crosswalk for Kubernetes](/docs/guides/crosswalk/kubernetes) that walk through how to go to production with EKS specifically, in addition to other managed Kubernetes offerings.
+
+Notice, for instance, that we didn't need to even provision any worker nodes. This is thanks to the `eks.Cluster` abstraction creating a default node pool for us, which has an auto-scaling policy that attempts to maintain the `desiredCapacity` while remaining within the bounds of `minSize` and `maxSize`. Sometimes you need more explicit control over the worker nodes, however, for reasons such as having more control over capacity, specializing the kind of compute or storage your workload needs, and so on. This ultimately devolves into managing EC2 instances by hand, which [the `eks.NodeGroup` class supports](https://www.pulumi.com/docs/guides/crosswalk/kubernetes/worker-nodes/).
+
+For example, this code creates a new EKS cluster, disabling the default node group &mdash; using `skipDefaultNodeGroup: true`, since we will create our own &mdash; and creates a single node group with an IAM role, desired instance type, scaling parameters, labels, and so on:
+
+```typescript
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+import * as eks from "@pulumi/eks";
+
+// Create a VPC for our cluster.
+const vpc = new awsx.ec2.Vpc("my-vpc");
+
+// IAM roles for the node group.
+const role = new aws.iam.Role("my-cluster-ng-role", {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+        Service: "ec2.amazonaws.com",
+    }),
+});
+let counter = 0;
+for (const policyArn of [
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+]) {
+    new aws.iam.RolePolicyAttachment(`my-cluster-ng-role-policy-${counter++}`,
+        { policyArn, role },
+    );
+}
```

Comment:
All this instanceprofile stuff isn’t needed is it?  @metral

Reply:
This is for the original, self-managed node groups, not the AWS managed ones.